wled00/FX.h

```diff
@@ -55,17 +53,17 @@
 /* each segment uses 52 bytes of SRAM memory, so if you're application fails because of
   insufficient memory, decreasing MAX_NUM_SEGMENTS may help */
 #ifdef ESP8266
-  #define MAX_NUM_SEGMENTS    12
+  #define MAX_NUM_SEGMENTS    18
```

Comment:
Definitely need a lot of testing here. I'm very doubtful that 18 segments + names + full `/json` request will fit in a 10k JSON buffer

Reply:
16 definitely does. And theoretically 18 does as well.

Reply:
If you agree we can lower the max number to 16 and recommend ESP32 for greater number of segments.
IMO 12 may be too low if someone wants to use a staircase usermod.

Reply:
Staircase is a very good usecase for this indeed. Just did some thourough testing and 16 segments seem to be a good limit. 16 named segments used 9187 bytes of JSON buffer, just barely fitting into the 9216 buffer. And there is also unpredictable buffer usage for usermod JSON. We might need to go with 10240 buffer size, I hope the extra 1K of allocation won't cause instabilities here.

What has me completely baffled though is that I'm observing a 200 b per _active_ segment higher memory usage even without segment names or effects using segment data. This is really strange, as segments do no dynamic allocation apart from name/data. Additionally, names seem to use about double the mem they should be (16 x 32-char long name uses ~1.2k, where it should only 512b).
My only explanation as to why this is happening is that the last network request is still buffered so that the serialized string length somehow matters (and a serialized segment string is 187 characters long, so that would be fitting). If that were the case, `/json`,`/json/si` and `/json/info` would all provide different values though, which does not happen.

Reply:
Just checked, this 200 b per seg behavior is also present in 0.13.0-b2. Thus not a new issue introduced by segment names.

Reply:
Thanks for your thorough testing and debugging.
I used 10k JSON buffer in the past but have reduced it to 9k when we did pre-serialize `col[]`.
Regarding the dynamic JSON buffer: I was thinking about serializing HTTP requests as to reduce chance of _out of memory_ conditions. What do you think?